modules/GroveLCD.js

```diff
@@ -0,0 +1,164 @@
+            // Clear the display completely if no args passed.
+            groveLCDAPI.i2cDevice.write(glcd.DISPLAY_ADDR,
+                                        new Buffer([0, glcd.CMD_SCREEN_CLEAR]));
+            return;
+        }
+        else if (col === undefined || row === undefined || numChar === undefined) {
+            // If any of the args undefined - print warning and return
+            console.log ("Missing args for GroveLCD.clear. Please pass col, row, numChar");
+            return;
+        }
+
+        // Move the cursor
+        this.resetCursor(col,row);
+
+        // Create blank space
+        var blanks = ' ';
+
+        for (var i = 0; i < numChar - 1; i++) {
+            blanks = blanks + ' ';
+        }
+
+        // Write blank space to LCD
+        this.writeText(blanks);
+    }
+
+    groveLCDAPI.resetCursor = function (col, row) {
+        // Reset the cursor position
+        row = row || 0;
+        col = col || 0;
+
+        if (row == 0) {
+            // Put the cursor on the first row at col
+            col |= 0x80;
+        } else {
+            // Put the cursor on the second row at col
+            col |= 0xC0;
+        }
+
+        this.i2cDevice.write(glcd.DISPLAY_ADDR,
+                             new Buffer([glcd.CMD_SET_DDRAM_ADDR, col]));
+    }
+
+    groveLCDAPI.setColor = function(colorObj) {
+        // Valid range for color is 0 - 255
+        // Write new color values
+        if (colorObj.blue && colorObj.blue > -1 && colorObj.blue < 256) {
+            this.i2cDevice.write(glcd.BACKLIGHT_ADDR,
+                                 new Buffer([glcd.REGISTER_B, blue]));
+        }
+        if (colorObj.red && colorObj.red > -1 && colorObj.red < 256) {
+            this.i2cDevice.write(glcd.BACKLIGHT_ADDR,
+                                 new Buffer([glcd.REGISTER_R, red]));
+        }
+        if (colorObj.green && colorObj.green > -1 && colorObj.green < 256) {
+            this.i2cDevice.write(glcd.BACKLIGHT_ADDR,
+                                 new Buffer([glcd.REGISTER_G, green]));
+        }
+    }
+
+
+    groveLCDAPI.writeText = function(word, col, row) {
+
+        if (col !== undefined && row !== undefined) {
+            // Move cursor to the specified location first
+            this.resetCursor(col, row);
+        }
+
+        var wordBuffer = new Buffer(word.length + 1);
+
+        // The first byte in the buffer is for the register address,
+        // which is where the word is going to be written to.
+        wordBuffer.writeUInt8(glcd.CMD_SET_CGRAM_ADDR, 0);
+
+        // Append the text we want to print after that
+        wordBuffer.write(word, 1);
+
+        // Write the buffer to I2C
+        this.i2cDevice.write(glcd.DISPLAY_ADDR, wordBuffer);
+    }
+
+    function init() {
+
+        // Clear the screen
+        groveLCDAPI.clear();
+
+        // Set our preferences for the Grove LCD
+        groveLCDAPI.i2cDevice.write(glcd.DISPLAY_ADDR,
```

Comment:
not sure you need to set up these initial display preferences here, since usually it's the app to decide at run time how they would like the display to be set up.